settings_dialog.js

```diff
@@ -79,14 +90,39 @@ function updateInstalledVersions(){
     });    
 }
 
+async function moveInstalledFiles(files, destination){
+    
+    movingFileModal.show();
+    let content = document.getElementById("movingFileModalContent");
+    content.innerHTML = "Moving files to: " + destination + " ...";
+    content.append(document.createElement("br"));
+    content.append(document.createTextNode("This may take several minutes, please be patient."));
+
+    await Promise.all(files.map(file =>
+        fs.move(file, path.join(destination, path.basename(file)))
+        .then(() => {
+            console.log("finished moving: " + path.basename(file));
+        })
+        .catch (err => {
```

Comment:
Yeah, that will be much more better.

Reply:
Wouldn't it work better if these .then and .catch would be moved to the Promise.all so that the first error of all of them is detected as well as the success is detected when all succeed.
Right now I think it's possible to trigger both the error and hit the successfully moved line.